.github/workflows/test.yml

```diff
@@ -183,10 +184,10 @@ jobs:
       - name: Test Integration
         env:
           NEXTEST_PROFILE: integration
-          INTEGRATION_TEST_SEQUENCER_VERSION: ${{matrix.version}}
+          INTEGRATION_TEST_SEQUENCER_VERSION: ${{ matrix.version }}
         run: |
-          cargo nextest run --archive-file nextest-archive.tar.zst  --verbose --no-fail-fast \
-          --workspace-remap $PWD
+          cargo nextest run --archive-file nextest-archive.tar.zst --verbose --no-fail-fast \
+          --workspace-remap $PWD $(if [ "${{ matrix.version }}" == "2" ]; then echo " smoke"; fi)
```

Comment:
I guess this is ok for now, but if we are going to run different tests for different targets matrix may not make much sense here anymore. If we are always going to have 2, then we can just have `base` job and `upgrade` job. They will still run in parallel. We can leave it like this for now though, I guess, and leave it for a later reorg PR.

Reply:
I think we will run upgrade test for every version other than initial (v2)